### PR TITLE
Chore/upgrade to newer laravel versions

### PR DESCRIPTION
**Pull request by ppelgrims**
Upgrade the codebase to support newer versions of both PHP and Laravel

### CODE REVIEW

File: .github/workflows/quality-assurance.yml [hunk 1]
```diff
@@ -34,16 +34,16 @@ jobs:
...
         env:
           CC_TEST_REPORTER_ID: ${{ secrets.CODECLIMATE_TEST_REPORTER_ID }}
         with:
+          coverageCommand: vendor/bin/phpunit --coverage-clover=coverage.clover
           coverageLocations: ${{github.workspace}}/*.clover:clover
           debug: true
 
       - name: Run PHPStan analysis
         run: vendor/bin/phpstan analyse --memory-limit=-1
-        if: ${{ matrix.php == '7.4' }}
```
Comment (remipelhate) [inline .github/workflows/quality-assurance.yml:49]: Why did you remove this? static analysis is typically done on a single version, as it is most often optimised for a single version as well.

File: .php-cs-fixer.dist.php [hunk 2]
```diff
@@ -7,6 +7,7 @@
     ->ignoreVCS(true);
 
 return (new PhpCsFixer\Config())
+    ->setRiskyAllowed(true)
```
Comment (remipelhate) [inline .php-cs-fixer.dist.php:10]: Which risky rules should be allowed exactly?

File: src/Events/MollieOrderWasUpdated.php [hunk 3]
```diff
@@ -13,20 +13,16 @@ final class MollieOrderWasUpdated implements MollieResourceStatusWasUpdated
...
-        $this->webhookCall = $webhookCall;
+    public function __construct(
+        /**
+         * @readonly
+         */
+        public OrderId $orderId,
+        /**
+         * @readonly
+         */
+        public WebhookCall $webhookCall
```
Comment (remipelhate) [inline src/Events/MollieOrderWasUpdated.php]: We're requiring at least PHP 8.2, so you can/should use proper readonly properties:
```suggestion
        public readonly OrderId $orderId,
        public readonly WebhookCall $webhookCall,
```

File: src/Events/MollieOrderWasUpdated.php [hunk 4]
```diff
@@ -13,20 +13,16 @@ final class MollieOrderWasUpdated implements MollieResourceStatusWasUpdated
...
+    ) {
     }
```
Comment (remipelhate) [inline src/Events/MollieOrderWasUpdated.php]: Should be on the same line. Please make sure to fix the CS fixer rules for this as well. You'll need to set `MultilinePromotedPropertiesFixer` and 
`ConstructorEmptyBracesFixer` from https://github.com/kubawerlos/php-cs-fixer-custom-fixers.
```suggestion
    ) {}
```

File: src/Events/MolliePaymentWasUpdated.php [hunk 5]
```diff
@@ -13,20 +13,16 @@ final class MolliePaymentWasUpdated implements MollieResourceStatusWasUpdated
...
+        /**
+         * @readonly
+         */
+        public PaymentId $paymentId,
+        /**
+         * @readonly
+         */
+        public WebhookCall $webhookCall
+    ) {
     }
```
Comment (remipelhate) [inline src/Events/MolliePaymentWasUpdated.php]:
```suggestion
    public function __construct(
        public readonly PaymentId $paymentId,
        public readonly WebhookCall $webhookCall,
    ) {}
```

File: src/Orders/WebhookCallOrderHistory.php [hunk 6]
```diff
@@ -16,11 +16,9 @@ final class WebhookCallOrderHistory implements OrderHistory
...
 
-    private LatestMollieWebhookCallByResourceId $latestMollieWebhookCallByResourceId;
-
-    public function __construct(LatestMollieWebhookCallByResourceId $latestMollieWebhookCallByResourceId)
-    {
-        $this->latestMollieWebhookCallByResourceId = $latestMollieWebhookCallByResourceId;
+    public function __construct(
+        private LatestMollieWebhookCallByResourceId $latestMollieWebhookCallByResourceId,
+    ) {
     }
```
Comment (remipelhate) [inline src/Orders/WebhookCallOrderHistory.php]:
```suggestion
    public function __construct(
        private LatestMollieWebhookCallByResourceId $latestMollieWebhookCallByResourceId,
    ) {}
```

File: src/Payments/WebhookCallPaymentHistory.php [hunk 7]
```diff
@@ -16,11 +16,9 @@ final class WebhookCallPaymentHistory implements PaymentHistory
...
 
-    private LatestMollieWebhookCallByResourceId $latestMollieWebhookCallByResourceId;
-
-    public function __construct(LatestMollieWebhookCallByResourceId $latestMollieWebhookCallByResourceId)
-    {
-        $this->latestMollieWebhookCallByResourceId = $latestMollieWebhookCallByResourceId;
+    public function __construct(
+        private LatestMollieWebhookCallByResourceId $latestMollieWebhookCallByResourceId,
+    ) {
     }
```
Comment (remipelhate) [inline src/Payments/WebhookCallPaymentHistory.php]:
```suggestion
    public function __construct(
        private LatestMollieWebhookCallByResourceId $latestMollieWebhookCallByResourceId,
    ) {}
```

File: src/Queries/LatestMollieWebhookCallByResourceId.php [hunk 8]
```diff
@@ -22,8 +21,8 @@ public function find(
             ->where('id', '!=', $ignoreWebhookCall->getKey())
             ->where('name', MollieSignatureValidator::NAME)
             ->where('payload', 'LIKE', "%\"id\":\"{$resourceId->value()}\"%")
-            ->when($payloadFragment, Closure::fromCallable([$this, 'filterByPayloadFragment']))
-            ->where(Closure::fromCallable([$this, 'ignoreFailedWebhookCalls']))
+            ->when($payloadFragment, ($this->filterByPayloadFragment(...))(...))
+            ->where(($this->ignoreFailedWebhookCalls(...))(...))
```
Comment (remipelhate) [inline src/Queries/LatestMollieWebhookCallByResourceId.php]: I think you can simply do:
```suggestion
            ->when($payloadFragment, $this->filterByPayloadFragment(...))
            ->where($this->ignoreFailedWebhookCalls(...))
```

File: src/Subscribers/SubscribeToMollieOrderStatusChanges.php [hunk 9]
```diff
@@ -14,20 +14,19 @@
...
-    {
-        $this->orders = $mollie->orders();
-        $this->events = $events;
-        $this->orderHistory = $orderHistory;
+    private readonly OrderEndpoint $orders;
+
+    public function __construct(
+        MollieApiClient $mollie,
+        private readonly Dispatcher $events,
+        private readonly OrderHistory $orderHistory
```
Comment (remipelhate) [inline src/Subscribers/SubscribeToMollieOrderStatusChanges.php]:
```suggestion
        private readonly OrderHistory $orderHistory,
```

File: src/Testing/Doubles/FakeMollieApiClient.php [hunk 10]
```diff
@@ -6,14 +6,21 @@
...
 final class FakeMollieApiClient extends MollieApiClient
 {
+    /**
+     * @throws ApiException
+     */
     public static function fake(Application $app): self
     {
-        return $app->instance(self::class, new self());
+        $mollie = new self();
+        $mollie->setApiKey($app['config']->get('mollie.key'));
```
Comment (remipelhate) [inline src/Testing/Doubles/FakeMollieApiClient.php]: Does the Mollie API client now fail if we don't set the ApiKey? Otherwise, I would avoid to do this, or try to use a fake key in order not to accidentally perform calls.

File: src/Testing/PrefixedResourceIdTestCase.php [hunk 11]
```diff
@@ -36,25 +36,27 @@ public function invalidMollieResourceIds(): Generator
...
+        $this::resourceIdClass()::fromString($value);
     }
 
-    public function validMollieResourceIds(): Generator
+    public static function validMollieResourceIds(): Generator
     {
-        yield 'Short identifier' => [$this->expectedPrefix() . Str::random(4)];
-        yield 'Long identifier' => [$this->expectedPrefix() . Str::random(8)];
+        yield 'Short identifier' => [fn (self $test): string => $test::expectedPrefix() . Str::random(4)];
+        yield 'Long identifier' => [fn (self $test): string => $test::expectedPrefix() . Str::random(8)];
```
Comment (remipelhate) [inline src/Testing/PrefixedResourceIdTestCase.php:48]: Why wrap the expectedPrefix generation in a closure? Can't we do this instead?
```suggestion
        yield 'Short identifier' => [self::expectedPrefix() . Str::random(4)];
        yield 'Long identifier' => [self::expectedPrefix() . Str::random(8)];
```

File: src/Testing/PrefixedResourceIdTestCase.php [hunk 12]
```diff
@@ -85,21 +91,23 @@ public function itCanReturnItsValue(string $value): void
...
+        $this->assertEquals($resourceId, unserialize($serializedResourceId, [$this::resourceIdClass()]));
     }
 
     /**
      * @test
      */
     public function itCanBeUsedAsAResourceId(): void
     {
-        $resourceId = $this->resourceIdClass()::fromString($this->expectedPrefix() . Str::random(8));
+        $resourceId = $this::resourceIdClass()::fromString($this::expectedPrefix() . Str::random(8));
```
Comment (remipelhate) [inline src/Testing/PrefixedResourceIdTestCase.php:100]:
```suggestion
        $resourceId = self::resourceIdClass()::fromString(self::expectedPrefix() . Str::random(8));
```

**Review by remipelhate — CHANGES_REQUESTED** [review]

**Comment by remipelhate** [comment]
@ppelgrims In addition to these changes, could you also use the new PHPUnit Test attributes instead of annotations? Feel free to do so in a separate PR, though.

File: src/Testing/PrefixedResourceIdTestCase.php (hunk 11 above)
Comment (ppelgrims) [reply src/Testing/PrefixedResourceIdTestCase.php:48]: I reckon the tests run for setting up the amount of tests that will run etc. before actually running the tests. It tries to call an abstract method which is not possible.

**Review by remipelhate — APPROVED** [review]
